### PR TITLE
[Flashscript] various fixes and small clean up.

### PR DESCRIPTION
**Pull request by andyblac**
this adds following:
[FlashScript] add python version check for script, as it requires ver 3.6.x
[FlashScript] add Shelly 1L support.
[FlashScript]  Shelly RGBW2: fix flashing Current Stock firmware.
[FlashScript] small cleanup.
[FlashScript] allow more time to test version, some stock device can take longer than 30 seconds.

### CODE REVIEW

File: tools/flash_shelly.py [hunk 1]
```diff
@@ -346,9 +360,12 @@ def parse_info(device_info, action, dry_run, silent_run, mode, exclude, version,
         homekit_dlurl = f'http://rojer.me/files/shelly/{version}/shelly-homekit-{model}.zip'
       break
 
-  stock_lfw = stock_release_info['data'][device_info['stock_model']]['version'].split('/v')[1].split('@')[0]
+  stock_lfw = stock_release_info['data'][stock_model]['version'].split('/v')[1].split('@')[0]
```
Comment (rojer) [inline tools/flash_shelly.py:363]: do not repeat yourself (DRY):

```
stock_model_info = stock_release_info['data'][stock_model]
# Give example of the version string you're expecting, it's hard to follow parsing code below.
stock_lfw = stock_model_info['version'].split('/v')[1].split('@')[0]

if not version:
  stock_dlurl = stock_model_info['url']
  if device_info['stock_model']  == 'SHRGBW2':
    stock_dlurl = stock_dlurl.replace('.zip', '-'+colour_mode+'.zip')
```

Comment (andyblac) [reply tools/flash_shelly.py]: ok, this should be cleaner https://github.com/mongoose-os-apps/shelly-homekit/pull/292/commits/2f287f4f89c4e5c68b2923fe50b65bb54e9ecb3d

File: tools/flash_shelly.py [hunk 2]
```diff
@@ -109,6 +109,10 @@ def get_info(host):
       info['model'] = shelly_model(info['app'])
     info['fw_type'] = 'homekit'
     info['fw_type_str'] = 'HomeKit'
+    if info['model']  == 'ShellyRGBW2':
```
Comment (rojer) [inline tools/flash_shelly.py:112]: this condition has the same code in both branches

File: tools/flash_shelly.py [hunk 3]
```diff
@@ -346,9 +367,13 @@ def parse_info(device_info, action, dry_run, silent_run, mode, exclude, version,
         homekit_dlurl = f'http://rojer.me/files/shelly/{version}/shelly-homekit-{model}.zip'
       break
 
-  stock_lfw = stock_release_info['data'][device_info['stock_model']]['version'].split('/v')[1].split('@')[0]
+  stock_model_info = stock_release_info['data'][stock_model]
+  stock_lfw = parseStockVersion(stock_model_info['version'])
   if not version:
-    stock_dlurl = stock_release_info['data'][device_info['stock_model']]['url']
+    if device_info['stock_model']  == 'SHRGBW2':
+      stock_dlurl = stock_model_info['url'].replace('.zip','-'+colour_mode+'.zip')
```
Comment (rojer) [inline tools/flash_shelly.py:374]: avoid using + for string formatting, use f""

**Review by rojer — CHANGES_REQUESTED** [review]

File: tools/flash_shelly.py (hunk 3 above)
Comment (andyblac) [reply tools/flash_shelly.py]: done

File: tools/flash_shelly.py (hunk 2 above)
Comment (andyblac) [reply tools/flash_shelly.py:112]: it a placeholder for when we do HomeKit RGBW2, i dont want to forget, i can comment out if you want ?

Comment (andyblac) [reply tools/flash_shelly.py]: tweaked https://github.com/mongoose-os-apps/shelly-homekit/pull/292/commits/624c66c05f4d9bfb5aa9837cf450ce55290416f0

**Review by rojer — APPROVED** [review]

**Comment by rojer** [comment]
thank you!

**Comment by andyblac** [comment]
no problem.
